lib/galaxy/tools/parser/output_collection_def.py

```diff
@@ -79,6 +81,17 @@ def __init__(self, **kwargs):
         self.directory = kwargs.get("directory", None)
         self.recurse = False
 
+    def to_dict(self):
+        return {
+            'discover_via': self.discover_via,
```

Comment:
discover_via is not part of the class DatasetCollectionDescription isn't it?

Reply:
The two concrete subclasses that are actually instantiated both define this I think. I can ``@six.add_metaclass(abc.ABCMeta)`` / ``@abc.abstractproperty`` if you'd like me to.